app/test/processing/app/CommandLineTest.java

```diff
@@ -94,8 +105,8 @@ public Process runArduino(boolean output, boolean success, File wd, String[] ext
 
     Process pr = rt.exec(args.toArray(new String[0]), null, wd);
     if (output) {
-      IOUtils.copy(pr.getInputStream(), System.out);
-      IOUtils.copy(pr.getErrorStream(), System.out);
+      consume(pr.getInputStream());
+      consume(pr.getErrorStream());
     }
     pr.waitFor();
```

Comment:
Should this also wait for the IO threads to be completed? I *think* this now has a race condition, where `pr.waitFor()` (and thus the testcase) completes before all the output has been consumed, which could lead to output being truncated or even be intermixed into the next testcase?

Reply:
And with the last commit, where these threads also run to capture output, I think this could even result in test failure.

Reply:
You're right, even if in practice this never happened, we need to join the two threads.
Will add a check for this.